shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1390,13 +1390,55 @@ void WebContents::HandleNewRenderFrame(
       static_cast<content::RenderWidgetHostImpl*>(rwhv->GetRenderWidgetHost());
   if (rwh_impl)
     rwh_impl->disable_hidden_ = !background_throttling_;
-
-  WebFrameMain::RenderFrameCreated(render_frame_host);
 }
 
 void WebContents::RenderFrameCreated(
     content::RenderFrameHost* render_frame_host) {
   HandleNewRenderFrame(render_frame_host);
+
+  auto* web_frame = WebFrameMain::FromRenderFrameHost(render_frame_host);
+  if (web_frame)
+    web_frame->Connect();
```

Comment:
Does this need to go in `HandleNewRenderFrame`? I don't think `RenderFrameCreated` is called for `window.open` and some other obscure cases.

Reply:
You're correct. I didn't notice the additional call to `HandleNewRenderFrame` from `WebContents::AddNewContents` initially. I'll move this back into `HandleNewRenderFrame`.